src/AppInstallerCLICore/COMContext.cpp

```diff
@@ -51,22 +51,33 @@ namespace AppInstaller::CLI::Execution
         Logging::SetExecutionStage(static_cast<uint32_t>(m_executionStage));
     }
 
-    void COMContext::SetLoggerContext(const std::wstring_view telemetryCorrelationJson, const std::string& caller)
+    void COMContext::SetContextLoggers(const std::wstring_view telemetryCorrelationJson, const std::string& caller)
     {
         m_correlationData = telemetryCorrelationJson;
-        Logging::Telemetry().SetTelemetryCorrelationJson(telemetryCorrelationJson);
-        Logging::Telemetry().SetCaller(caller);
-        Logging::Telemetry().LogStartup(true);
+
+        std::unique_ptr<AppInstaller::ThreadLocalStorage::PreviousThreadGlobals> setThreadGlobalsToPreviousState;

```

Comment:
Two comments here:
1. It is better to use `auto` for these RAII lifetime holders as you aren't ever going to interact with them directly.  That allows the return type to change tomorrow without updating every call site.
2. You are in the type that owns the TLS object itself; there is no need to set the TLS pointer then turn around and retrieve it back.  I get that you are also reusing `SetLoggers` to initialize the global state, but I would not be shocked to have that diverge in the future.  And there are ways to reuse that code without using the TLS state.

Neither of these necessitates a change, but they are things to think about for the future.